zk/dnshostprovider.go

```diff
@@ -34,17 +34,19 @@ func (hp *DNSHostProvider) Init(servers []string) error {
 	for _, server := range servers {
 		host, port, err := net.SplitHostPort(server)
 		if err != nil {
-			return err
+			// If one node is down, Init should still continue
+			continue
 		}
 		addrs, err := lookupHost(host)
 		if err != nil {
-			return err
+			// If one node is down, Init should still continue
```

Comment:
I like this change

Reply:
Thanks @jdef for reviewing. I have updated this diff